### PR TITLE
Add java 17

### PR DESCRIPTION
**Pull request by timja**

### CODE REVIEW

File: updatecli/updatecli.d/jdk17.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    name: Get the latest Adoptium JDK17 version
+    spec:
+      owner: "adoptium"
+      repository: "temurin17-binaries"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      versionFilter:
+        kind: regex
+        # jdk-17.0.12+7(https://github.com/adoptium/temurin17-binaries/releases/tag/jdk-17.0.12%2B7) is OK
+        pattern: "^jdk-17+(\\d*)$"
```
Comment (timja) [inline updatecli/updatecli.d/jdk17.yml:16]: not 100% sure I did the right thing here

File: updatecli/updatecli.d/jdk17.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    kind: githubRelease
+    name: Get the latest Adoptium JDK17 version
+    spec:
+      owner: "adoptium"
+      repository: "temurin17-binaries"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      versionFilter:
+        kind: regex
+        # jdk-17.0.12+7(https://github.com/adoptium/temurin17-binaries/releases/tag/jdk-17.0.12%2B7) is OK
```
Comment (dduportal) [inline updatecli/updatecli.d/jdk17.yml:15]:
```suggestion
        # jdk-17+35(https://github.com/adoptium/temurin17-binaries/releases/tag/jdk-17%2B35) is OK
```

File: updatecli/updatecli.d/jdk17.yml (hunk 1 above)
Comment (dduportal) [reply updatecli/updatecli.d/jdk17.yml:16]:
```suggestion
        pattern: "^jdk-17\\+(\\d*)$"
```

The character `+` must be escaped in the regex to be treated as a litteral

**Review by dduportal — CHANGES_REQUESTED** [review]

**Comment by dduportal** [comment]
Sounds like the Windows script is failing: https://infra.ci.jenkins.io/blue/organizations/jenkins/packer-images/detail/PR-95/3/pipeline/182#step-189-log-103 , which does not happen on the principal branch (https://infra.ci.jenkins.io/blue/organizations/jenkins/packer-images/detail/main/18/pipeline).

**Review by dduportal — APPROVED** [review]
LGMT good job!

**Pull request merged**